src/nvim/tui/tui.c

```diff
@@ -592,37 +612,57 @@ static void tui_terminal_after_startup(TUIData *tui)
   flush_buf(tui);
 }
 
-/// Stop the terminal but allow it to restart later (like after suspend)
-static void tui_terminal_stop(TUIData *tui)
+void tui_error_exit(TUIData *tui, Integer status)
+  FUNC_ATTR_NONNULL_ALL
+{
+  tui->seen_error_exit = (int)status;
+}
+
+void tui_stop(TUIData *tui)
+  FUNC_ATTR_NONNULL_ALL
 {
   if (uv_is_closing((uv_handle_t *)&tui->output_handle)) {
     // Race between SIGCONT (tui.c) and SIGHUP (os/signal.c)? #8075
     ELOG("TUI already stopped (race?)");
     tui->stopped = true;
     return;
   }
-  tinput_stop(&tui->input);
-  // Position the cursor on the last screen line, below all the text
-  cursor_goto(tui, tui->height - 1, 0);
-  terminfo_stop(tui);
-}
 
-void tui_error_exit(TUIData *tui, Integer status)
-{
-  tui->seen_error_exit = (int)status;
-}
+  tui->input.callbacks.primary_device_attr = tui_stop_cb;
+  terminfo_disable(tui);
+
+  // Wait until DA1 response is received
+  LOOP_PROCESS_EVENTS_UNTIL(tui->loop, tui->loop->events, EXIT_TIMEOUT_MS, tui->stopped);
```

Comment:
Maybe as a followup, if we continue to get reports, we could print something like this if the timeout is reached:

    Nvim: timeout waiting for DA1 response